src/nwb_conversion_tools/tools/roiextractors/roiextractors.py

```diff
@@ -212,76 +213,87 @@ def get_nwb_imaging_metadata(imgextractor: ImagingExtractor):
+    nwbfile: NWBFile, optional
+        If passed, this function will fill the relevant fields within the NWBFile object.
+        E.g., calling
+            write_recording(recording=my_recording_extractor, nwbfile=my_nwbfile)
         will result in the appropriate changes to the my_nwbfile object.
-    metadata: dict
-        metadata info for constructing the nwb file (optional).
-    overwrite: bool
-        If True and save_path is existing, it is overwritten
+        If neither 'save_path' nor 'nwbfile' are specified, an NWBFile object will be automatically generated
+        and returned by the function.
+    metadata: dict, optional
+        Metadata dictionary with information used to create the NWBFile when one does not exist or overwrite=True.
+    overwrite: bool, optional
+        Whether or not to overwrite the NWBFile if one exists at the nwbfile_path.
+        The default is False (append mode).
+    verbose: bool, optional
+        If 'nwbfile_path' is specified, informs user after a successful write operation.
+        The default is True.
     num_chunks: int
         Number of chunks for writing data to file
     use_times: bool (optional, defaults to False)
         If True, the times are saved to the nwb file using imaging.frame_to_time(). If False (defualt),
         the sampling rate is used.
     """
     assert save_path is None or nwbfile is None, "Either pass a save_path location, or nwbfile object, but not both!"
-
     if nwbfile is not None:
         assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
-    # Update any previous metadata with user passed dictionary
+
+    # TODO on or after August 1st, 2022, remove argument and deprecation warnings
+    if save_path is not None:
+        will_be_removed_str = "will be removed on or after August 1st, 2022. Please use 'nwbfile_path' instead."
+        if nwbfile_path is not None:
+            if save_path == nwbfile_path:
+                warn(
+                    "Passed both 'save_path' and 'nwbfile_path', but both are equivalent! "
+                    f"'save_path' {will_be_removed_str}",
+                    DeprecationWarning,
+                )
+            else:
+                warn(
+                    "Passed both 'save_path' and 'nwbfile_path' - using only the 'nwbfile_path'! "
+                    f"'save_path' {will_be_removed_str}",
+                    DeprecationWarning,
+                )
+        else:
+            warn(
+                f"The keyword argument 'save_path' to 'spikeinterface.write_recording' {will_be_removed_str}",
+                DeprecationWarning,
+            )
+            nwbfile_path = save_path
+
     if metadata is None:
         metadata = dict()
     if hasattr(imaging, "nwb_metadata"):
         metadata = dict_deep_update(imaging.nwb_metadata, metadata)
     metadata = dict_deep_update(get_nwb_imaging_metadata(imaging), metadata)
-    if nwbfile is None:
```

Comment:
So nice that we can rid of all of this : D

Reply:
These functions haven't been looked at in a while, so I'm sure there's a lot more that can be improved lol